proposals/Exceptions.md

```diff
@@ -736,18 +736,24 @@ the event indices (Used for both imports and module-defined).
 
 ## Comparisons of the two proposals
 
-- Proposal 2 is more expressive and possibly provides more flexibility
-  for future frontend developers for other languages.
-
-- As Proposal 2 introduces first-class exception reference type, we have to
-  manage lifetime of exception objects, from which arises several questions.
+- Proposal 2 introduces a first-class exception reference type. This raises several
+  questions about exception lifetime management:
   - How do we manage exception objects' lifetime in non-GC embeddings? Do we
     make reference counting mandatory?
   - Who is responsible for deleting exception objects?
   - What should we do for except_ref values of invalid exception objects already
     deleted?
   - How should exception reference type be related to the existing reference
     type or GC proposal?
+   
+  Consequently, Proposal 1 would be simpler to implement for VMs which do not need
+  reference types or related functionality such as GC objects.
+  
+- The first-class exception type makes Proposal 2 more expressive, possibly providing
+  more flexibility for frontend developers for non-C langauges. In particular, allowing
+  exception objects to escape catch blocks may simplify control flow translation.
+  Conversely, it is slightly more complex for languages which do not have convenient ways to model
```

Comment:
Not sure I follow this. The source language wouldn't need to be able to represent reference types, they would only be introduced by lowering. Moreover, variant 1 can actually be desugared directly into variant 2. Hence a producer could completely ignore the difference between the variants if it wants to.

Reply:
If you want to pass the exception package into a library function, it would be nice to be able to implement that library function in the source language, so you'd want a way to represent it. It's not impossible, just more awkward.